docs/developer-guide/working-with-custom-formatters.md

```diff
@@ -67,9 +68,11 @@ the list of messages for `errors` and/or `warnings`.
 The following are the fields of the result object:
 
 - **filePath**: The path to the file relative to the current working directory (the path from which eslint was executed).
-- **messages**: An array of message objects. See below for more info about messages
-- **errorCount**: The number of errors for the given file
-- **warningCount**: the number of warnings for the give file
+- **messages**: An array of message objects. See below for more info about messages.
+- **errorCount**: The number of errors for the given file.
+- **warningCount**: The number of warnings for the given file.
+- **source**: The source code for the given file. This property is omitted if this file has no errors/warnings or if the `output` property is present.
+- **output**: The source code for the given file with as many fixes applied as possible. This property is omitted if no fix is available.
```

Comment:
I'm not sure where, but we should probably mention that the `source` property on the individual messages will be going away and that people should instead use `source` or `output` from the top-level of the structure.